rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/InfrastructureManager.java

```diff
@@ -601,34 +604,40 @@ protected void shutDown() {
     }
 
     /**
-     * First fetch the {@link NodeSourceData} associated to this infrastructure, then update the information related to
-     * the infrastructure, and then update in database the {@link NodeSourceData}.
+     * If nodes recovery is enabled, persist the current state of the node
+     * source.
+     * If the node source data of this infrastructure is not know, it is first
+     * fetched in database.
+     * The node source data is never persisted if the current infrastructure is
+     * the {@link NodeSource#DEFAULT_LOCAL_NODES_NODE_SOURCE_NAME}.
      */
     public void persistInfraVariables() {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 16 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)